controllers/remote/cluster_cache_tracker.go

```diff
@@ -335,11 +348,20 @@ func (t *ClusterCacheTracker) newClusterAccessor(ctx context.Context, cluster cl
 		return nil, err
 	}
 
+	// Generating a new private key to be used for generating temporary certificates for used for connecting to
+	// etcd on the target cluster.
+	// NOTE: Generating a private key is an expensive operation, so we store it in the cluster accessor.
+	etcdKey, err := certs.NewPrivateKey()
+	if err != nil {
+		return nil, err
```

Comment:
```suggestion
		return nil, errors.Wrapf(err, "error creating etcd client key for remote cluster %q", cluster.String())
```
nit: I fixed the other errors in my CR bump PR to end up with nicely wrapped errors